spec/lang/intrinsic.md

```diff
@@ -7,8 +7,7 @@ impl<M: Memory> Machine<M> {
         &mut self,
         intrinsic: Intrinsic,
         arguments: List<Value<M>>,
-        ret_place: Option<Place<M>>,
-    ) -> NdResult;
+    ) -> NdResult<Option<(Value<M>, Type)>>;
```

Comment:
Why is there an `Option` here?

If an intrinsic returns `()`, that can be represented as an empty tuple.

Reply:
This would be an option. I don't yet see how this would make the code more understandable, since it would always need `ret((Value::Tuple(list![]), Type::Tuple{fields:list![], size: Size::ZERO}))`. I tried just creating constants, but `Value` is dependent on M and both need an empty list.

Reply:
This works. But I have changed the behavior of intrinsics. So the test `ub::alloc::alloc_noret` fails.

Reply:
That's just a changed error message it seems? Seems fine to me.

Reply:
No. Since we now can't distinguish between intrinsics that return something and those who don't (at least it doesn't do that currently). It is now allowed to call `alloc` without a place to put the pointer into. This was not the case so far.

Reply:
Okay I looked closer into the error message. The issue is, since the test assumes the alloc to never return, It does not specify the next block to be executed. The UB we currently get is because of this missing next block. Not because the allocation failed. This is because the interpreter newly allows for `alloc` without a return place. I would need to completely remove this test to pass the checks.

Reply:
Yeah seems fine to remove that test.